h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/ProxyStarter.java

```diff
@@ -0,0 +1,78 @@
+package water;
+
+import water.init.HostnameGuesser;
+import water.init.NetworkInit;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.UnknownHostException;
+
+public class ProxyStarter {
+
+  public static String start(String[] args, JettyProxy.Credentials credentials, String proxyTo) {
+    if (! proxyTo.endsWith("/"))
+      proxyTo = proxyTo + "/";
+
+    H2O.BaseArgs baseArgs = H2O.parseH2OArgumentsTo(args, new H2O.OptArgs());
+
+    JettyProxy proxy = initializeProxy(baseArgs, credentials, proxyTo);
+
+    InetAddress address = HostnameGuesser.findInetAddressForSelf(baseArgs.ip, baseArgs.network);
+    return H2O.getURL(proxy.getScheme(), address, proxy.getPort(), baseArgs.context_path);
+  }
+
+  private static JettyProxy initializeProxy(H2O.BaseArgs args, JettyProxy.Credentials credentials, String proxyTo) {
+    int proxyPort = args.port == 0 ? args.baseport : args.port;
+
+    JettyProxy proxy = new JettyProxy(args, credentials, proxyTo);
+
+    // PROXY socket is only used to find opened port on given ip
+    ServerSocket proxySocket = null;
+
+    while (true) {
+      try {
+        proxySocket = args.web_ip == null ? // Listen to any interface
+                new ServerSocket(proxyPort) : new ServerSocket(proxyPort, -1, getInetAddress(args.web_ip));
+        proxySocket.setReuseAddress(true);
+
+        // race condition: another process can use the port while proxy is starting
+        proxySocket.close();
+        proxy.start(args.web_ip, proxyPort);
+
+        break;
+      } catch (Exception e) {
+        System.err.println("TRACE: Cannot allocate API port " + proxyPort + " because of following exception: " + e.getMessage());
```

Comment:
Right, h2odriver itself doesn't use Logging infrastructure. At this point no logger is initialized/used.